enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,511 @@
+`infrastructure` API if they need to configure their operand
+differently based on the high-availability mode.
+
+### Risks and Mitigations
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+#### Control plane rollouts
+
+kube-apiserver is periodically reconfigured by its operator in
+response to changes in cluster state (e.g. weekly rotation of etcd
+encryption keys). When this occurs, the apiserver may be inaccessible
+for up to 2 minutes. Workloads that do not depend on the apiserver
+would be expected to run without issue during this interval. Workloads
+that do depend on apiserver availability would need to be resilient to
+these events. OpenShift core components are already resilient in this
+way.
+
+#### Cluster Machine Approver
+
+Auto-approval of certificate signing requests requires 2 sources of
+truth to avoid security attacks like
+[kubeletmein](https://github.com/openshift/machine-config-operator/issues/731). In
+single-node deployments we do not have a second source of truth, and
+need to disable the machine-approver-operator. An outside tool can be
+used to approve any certificate signing requests instead.
+
+#### Lack of high-availability
+
+In a single-node configuration OpenShift would have limited ability to
+provide resiliency features. We will need to clearly communicate these
+limitations to customers choosing to use single-node deployments.
+
+#### Failure to reboot
+
+It is possible that a single node may fail to reboot due to an issue
+with configuration, hardware, or environment. These single-node
+deployments will be managed by external tools, and as with other
+failures related to providing resilient operation, host boot failures
+will need to be detected and mitigated by those tools, rather than the
+local OpenShift instance.
+
+#### Reboots for reconfiguration
+
+Some reconfiguration operations will unavoidably cause the node to
+reboot. These tasks will need to be scheduled during maintenance
+windows.  Ideally we will be able to rely on the [graceful shutdown
+support in
+kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
+to stop all workloads safely as part of the reboot.
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+2. How do we hand off ownership of the `etcd-quorum-guard` Deployment
+   from the `cluster-version-operator` to the `cluster-etcd-operator`?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have CI
+coverage informing the release. An end-to-end job using a single-node
+deployment and running an appropriate subset of the standard OpenShift
+tests will be created and configured to block accepting release images
```

Comment:
How will we define the subset?  This might take more work than expected..

Reply:
@romfreiman @eranco74 can we link to the CI job definition that you've already created?

I would expect the test suite to include all of the tests except those that use features explicitly not enabled (upgrades) or that check things like how many replicas of a service are running (and those should be fixed to look at the new API to decide what the right answer is).

Reply:
I've added links to the PRs defining the new jobs in the latest draft.